### PR TITLE
SNOW-787565: Use dynamic scaling thread pool instead of fixed thread pool

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**
We want to release the resources in case of an idle client

### CODE REVIEW

**Comment by sfc-gh-tzhang** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by sfc-gh-tzhang** [comment]
recheck

File: src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java [hunk 1]
```diff
@@ -319,7 +320,13 @@ private void createWorkers() {
                 * (1 + this.owningClient.getParameterProvider().getIOTimeCpuRatio()),
             MAX_THREAD_COUNT);
     this.buildUploadWorkers =
-        Executors.newFixedThreadPool(buildUploadThreadCount, buildUploadThreadFactory);
+        new ThreadPoolExecutor(
+            1,
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java:324]: nit: /* corePoolSize*/ after a constant 1

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm

**Pull request merged**